esrally/tracker/index.py

```diff
@@ -46,10 +46,19 @@ def update_index_setting_parameters(settings):
             settings[s] = param.format(orig=orig_value)
 
 
+allowed_list_hidden_indices = [
```

Comment:
Since those indices are specific to your use case, let's not hardcode them in Rally, they should be specified in the command line, which would be more explicit and won't surprise other users.

Reply:
I agree let me provide it like that

Reply:
Done @pquentin ,

See update run with no ALLOWED_LIST:

 esrally create-track --track=test --target-hosts=test.rally.gizas.elastic-observability-ocp.elastic.dev:9200 --client-options="use_ssl:true,verify_certs:false,ca_certs:false,basic_auth_user:'elastic',basic_auth_password:'C7GfaRy9E83E950v74IwPZq6'" --data-streams="metrics-*" --output-path=~/tracks

    ____        ____
   / __ \____ _/ / /_  __
  / /_/ / __ `/ / / / / /
 / _, _/ /_/ / / / /_/ /
/_/ |_|\__,_/_/_/\__, /
                /____/

[INFO] Connected to Elasticsearch cluster [ha-es-zone-d-0] version [8.3.0-SNAPSHOT].

Extracting documents for index [.ds-metrics-elastic_ag...     1000/1000 docs [100.0% done]
Extracting documents for index [.ds-metrics-elastic_ag...     5911/257181 docs [2.3% done]^C[INFO] Aborted create-track.
[output ommitted ...]

Reply:
Thanks, this is great. For better consistency, we should also accept hidden indices when specified directly with `--indices`. And let's name the parameter that you pass to every function "allow_hidden_indices" instead of "datastream_flag", this will be clearer.

Reply:
See comment https://github.com/elastic/rally/pull/1531#issuecomment-1178911028
For latest status